src/main/cluster-detectors/detector-registry.ts

```diff
@@ -5,11 +5,17 @@
 
 import { observable } from "mobx";
 import type { ClusterMetadata } from "../../common/cluster-types";
-import { Singleton } from "../../common/utils";
 import type { Cluster } from "../../common/cluster/cluster";
 import type { BaseClusterDetector, ClusterDetectionResult } from "./base-cluster-detector";
+import type { K8sRequest } from "../k8s-request.injectable";
+
+interface Dependencies {
+  k8sRequest: K8sRequest;
+}
+
+export class DetectorRegistry {
+  constructor(private dependencies: Dependencies) {}
```

Comment:
nit: dependencies should always be marked as `readonly` and basically always be `protected` and not `private` (if not for consistency but also so that child classes can extend the dependencies if need be).

Reply:
I'd like to argue that there shouldn't be child classes at all. `private`, `protected` etc. lose their meaning when we start composing stuff instead of inherit. 

Composition over inheritance.